### PR TITLE
Azcore v1.8.0

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This contains the following cherry-picks.

4a66b4f8eb8cdeb173a750c53069edbc74d3520d
5624b2d924f5c5d5d42e8c3f7ed914755b25b2f8
f839de425de7ed3a53cc96eba7ecdf3c42af70c2
9b73d1f0bc301cb2f9827ae16b9c1d4d66b46756
f6a43aafb29dc67b5c989d9fc15b8dd7b494e50c
f9574cdbffbd45bb995beadc36785faec776032d
96a6b852c44a7e6c94e7b524add087c4db1d6b7b
081c5e3c5bc3ca4e93179dbc19c5bff373a84049
cd1cd878489e5223b3d70f9279dd4a8f2194688e
5d0404765381c7ce294ce674337591dec8e34107
0e745398c3cf95eb4af79c8cfb995a64cecc0ae5
8db55a4db15596465292549169a0f4ccfb112e62
10f029c10f513c788ff95ead6f2509f82b90ee50
81c4f033af4da4a92cac408a10c9d7f842140182

### CODE REVIEW

File: sdk/azcore/arm/runtime/policy_register_rp.go [hunk 1]
```diff
@@ -176,6 +176,7 @@ var unregisteredRPCodes = []string{
 	"MissingSubscriptionRegistration",
 	"MissingRegistrationForResourceProvider",
 	"Subscription Not Registered",
+	"SubscriptionNotRegistered",
```
Comment (RickWinter) [inline sdk/azcore/arm/runtime/policy_register_rp.go:179]: Why are there both of these?

File: sdk/azcore/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
 # Release History
 
+## 1.8.0 (2023-10-05)
+
+### Features Added
+
+* `messaging/CloudEvent` allows you to serialize/deserialize CloudEvents, as described in the CloudEvents 1.0 specification: [link](https://github.com/cloudevents/spec)
+* Added `TokenRequestOptions.EnableCAE` to indicate whether to request a CAE token
```
Comment (jhendrixMSFT) [inline sdk/azcore/CHANGELOG.md:8]: @chlowell is this the only changelog note needed for CAE?

File: sdk/azcore/messaging/cloud_event.go [hunk 3]
```diff
@@ -0,0 +1,291 @@
+// Copyright 2017 Microsoft Corporation. All rights reserved.
+// Use of this source code is governed by an MIT
+// license that can be found in the LICENSE file.
```
Comment (RickWinter) [inline sdk/azcore/messaging/cloud_event.go:3]: This copyright is non-standard with the year.

File: sdk/azcore/CHANGELOG.md (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/azcore/CHANGELOG.md:8]: Hmm looks like these notes should also be added.

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/CHANGELOG.md#features-added-10

Let me know if there are others.

**Review by RickWinter — COMMENTED** [review]
Has the surface area additions been approved by Jeff?

**Comment by jhendrixMSFT** [comment]
The public surface area additions have been approved in their constituent PRs but will require a formal approval in APIView before we can release.

CC @JeffreyRichter

File: sdk/azcore/arm/runtime/policy_register_rp.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azcore/arm/runtime/policy_register_rp.go:179]: Per the ARM spec, the value is supposed to _always_ be `MissingSubscriptionRegistration`.  Unfortunately, some RPs got it wrong (and can't fix it due to back-compat) which is why we now have four values.

File: sdk/azcore/messaging/cloud_event.go (hunk 3 above)
Comment (jhendrixMSFT) [reply sdk/azcore/messaging/cloud_event.go:3]: Good catch.  I'll fix this in `main` then cherry-pick the commit.

Comment (jhendrixMSFT) [reply sdk/azcore/messaging/cloud_event.go:3]: https://github.com/Azure/azure-sdk-for-go/pull/21695

File: sdk/azcore/CHANGELOG.md (hunk 2 above)
Comment (chlowell) [reply sdk/azcore/CHANGELOG.md:8]: Cross-tenant ARM and multitenant auth are already GA, so the new stuff for this release v1.8.0 is:
- Added `Claims` and `EnableCAE` fields to `TokenRequestOptions`
- ARM bearer token policy handles CAE challenges

**Review by chlowell — APPROVED** [review]

File: sdk/azcore/CHANGELOG.md (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/azcore/CHANGELOG.md:10]: Fixed.

File: sdk/azcore/messaging/cloud_event.go (hunk 3 above)
Comment (jhendrixMSFT) [reply sdk/azcore/messaging/cloud_event.go]: Fixed.

**Comment by jhendrixMSFT** [comment]
@benbp are there other commits I should cherry-pick into this branch to get CI working?

File: sdk/azcore/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,32 @@
 # Release History
 
+## 1.8.0 (2023-10-05)
+
+### Features Added
+
+* Added `Claims` and `TenantID` fields to `policy.TokenRequestOptions`.
```
Comment (chlowell) [inline sdk/azcore/CHANGELOG.md:7]: `TenantID` was in a prior stable version
```suggestion
* Added `Claims` and `EnableCAE` fields to `policy.TokenRequestOptions`.
```

Comment (jhendrixMSFT) [reply sdk/azcore/CHANGELOG.md]: I can't copy/paste :p.  Fixed (also removed the duplicate entry).

**Comment by jhendrixMSFT** [comment]
@benbp is there an easy way for me to pull in all of `eng` from `main`?

**Comment by benbp** [comment]
@jhendrixMSFT I think something like this will work. Replace `origin` with whatever your upstream remote is named.

```
git fetch --all
git merge --allow-unrelated-histories -s ours --no-commit origin/main
git rm -r eng/
git read-tree --prefix=eng/ -u origin/main:eng
git commit -m 'Merge eng from main'
```

**Comment by jhendrixMSFT** [comment]
@benbp it's working now, thanks so much for the help.

**Comment by benbp** [comment]
Awesome!

**Comment by jhendrixMSFT** [comment]
https://github.com/Azure/azure-sdk-for-go/issues/21116 strikes again

**Pull request merged**